policy-controller/grpc/src/outbound.rs

```diff
@@ -251,20 +265,9 @@ fn to_service(
             }),
         });
 
-        let mut http_routes = outbound.http_routes.into_iter().collect::<Vec<_>>();
-        let mut grpc_routes = outbound.grpc_routes.into_iter().collect::<Vec<_>>();
-
-        if !grpc_routes.is_empty() {
-            grpc_routes.sort_by(timestamp_then_name);
-            grpc::protocol(
-                backend,
-                grpc_routes.into_iter(),
-                accrual,
-                outbound.grpc_retry,
-                outbound.timeouts,
-                allow_l5d_request_headers,
-            )
-        } else {
+        // if we have no explicit routes attached to the parent, always attempt protocol detection
+        if no_explicit_routes(&outbound) {
```

Comment:
How does this compare to the current approach of just starting with the most specific route type and falling back to less and less specific?  ie if there are grpcRoutes, use those, else if there are httproutes, use those, else if there are tls routes, use those, else if there are tcp routes, use those, ELSE do http detect.

Reply:
good point. My approach is essentially the same but more complicated. Will incorporate your suggestion